chatgpt_vs_claude.py

```diff
@@ -69,36 +83,36 @@ def preprocess_message(message: ProtocolMessage, bot: str) -> ProtocolMessage:
 
 
 def preprocess_query(query: QueryRequest, bot: str) -> QueryRequest:
+    """Parses the two bot responses and keeps the one for the current bot."""
     new_query = query.model_copy(
         update={"query": [preprocess_message(message, bot) for message in query.query]}
     )
     return new_query
 
 
+async def stream_request_wrapper(
+    query: QueryRequest, bot: str
+) -> AsyncIterator[PartialResponse]:
+    """Wraps stream_request and labels the bot response with the bot name."""
+    yield PartialResponse(text=f"**{bot.title()}** says:\n")
+    async for msg in stream_request(
+        preprocess_query(query, bot), bot, query.access_key
+    ):
+        if isinstance(msg, Exception):
+            yield PartialResponse(
+                text=f"**{bot.title()}** ran into an error", is_replace_response=True
+            )
+            return
+        yield msg
```

Comment:
This will behave incorrectly if the underlying bot yields a replace response message: the "ChatGPT says:" part will be replaced away. We need to check here for is_replace_response and if it is true, re-send the "ChatGPT says:" intro.

Reply:
Should have been addressed now.

Reply:
No, this is still wrong, as the `yield msg` will still replace away the "label" message.

Reply:
ah, mb, forgot to change that part.